src/cio_websocket.c

```diff
@@ -149,10 +127,9 @@ static void send_frame(struct cio_websocket *ws, struct cio_write_buffer *payloa
 	ws->wbh = payload;
 	add_websocket_header(ws);
 	ws->user_write_handler = written_cb;
-	ws->ws_flags.writing_frame = 1;
 	enum cio_error err = ws->bs->write(ws->bs, payload, write_complete, ws);
 	if (err != CIO_SUCCESS) {
-		ws->ws_flags.to_be_closed = 1;
+		//TODO: call callback
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=c%3AS1135)